File: sources/scalac/transformer/matching/NoSeqVariableTraverser.java
package scalac.transformer.matching ;

import scalac.util.Name ;
import scalac.ast.Tree ;
import scalac.ast.Traverser ;
import scalac.symtab.Symbol ;

import java.util.Vector;

class NoSeqVariableTraverser extends CollectVariableTraverser {

      public void traverse(Tree tree) {
            switch (tree) {
            case Sequence(_):
                  return ;
            default:
                  super.traverse( tree );
            }
      }


      public NoSeqVariableTraverser() {
            super();
      }

      static Vector varsNoSeq( Tree  pat ) {

            NoSeqVariableTraverser nvt = new NoSeqVariableTraverser();
            nvt.traverse( pat );
            return nvt.vars;

      }

      static Vector varsNoSeq( Tree[]  pats ) {

            NoSeqVariableTraverser nvt = new NoSeqVariableTraverser();
            for(int i = 0; i < pats.length; i++)
                  nvt.traverse( pats[ i ] );
            return nvt.vars;

      }

}

File: sources/scala/tools/scalac/transformer/matching/NoSeqVariableTraverser.scala


import scalac.util.Name ;
import scalac.ast.Tree ;
import scalac.ast.Traverser ;
import scalac.symtab.Symbol ;

import java.util.HashSet;

package scala.tools.scalac.transformer.matching {

class NoSeqVariableTraverser extends CollectVariableTraverser {

      override def traverse(tree:Tree ): Unit = {
        tree.match {
          case Tree.Sequence(_) =>
            return ;
          case _ =>
            super.traverse( tree );
        }
      }

  def varsNoSeq(pat: Tree):HashSet = {
    vars.clear();
    traverse( pat );
    return vars;

  }

  def varsNoSeq(pats: Array[Tree]): HashSet = {

    //NoSeqVariableTraverser nvt = new NoSeqVariableTraverser();
    var i = 0;
    while(i < pats.length) {
      traverse( pats( i ) );
      i = i + 1;
    }
    vars;

  }

}
}
